docs/DeveloperGuide.md

```diff
@@ -1,34 +1,357 @@
+
+
+#### 3.2.3. Cheat Sheet Structure (Aldo)
+
+To further understand our cheatsheet management application, it is important to understand the inner workings of the CheatSheet class. 
+CheatLogs comes with preloaded cheat sheets and the application has an in-built ability for the user to add, edit, and delete their own cheat sheets. 
+The preloaded cheat sheet cannot be modified by the user, as the user-made cheatsheet is able to be modified by the user.
+
+
+The CheatSheet class is the class that models the cheat sheet that we used in our program. 
+Each of our cheatsheet object is intended to store one cheatsheet topic for one specific programming language (both preloaded and user-created cheat sheets).
+
+This system is intended to increase user access and organization of the cheat sheets.
+Each of the CheatSheet object has:
+1. 	Name
+2. 	Programming Language
+3. 	Description (Content of the cheatsheet)
+4. 	isFavorite
+5. 	isModifiable
+6. 	Date Created
+7. 	Date Modified
+
+![](Images/Image8.PNG)
+
+Image 6: class diagram of the cheatsheet class
+
+##### Development Goals
+
+We are trying to move our existing cheatsheet file format from txt to xml. 
+XML files provide inbuilt tags to differentiate and classify members of the document. 
+With the use of XML file, it simplifies the parsing process of the cheatsheet files into the cheatsheet class, and vice versa.
+
+![](Images/Image9.PNG)
+
+Image 7: XML file showing the content of a cheatsheet file
+
+#### 3.2.4. Cheat Sheet Management (Adhy)
+
+All cheat sheets, both pre-loaded and user-defined, are stored in one static class called CheatSheetList during runtime. 
+Upon receiving a valid input from the user, the execute() method from Command class will invoke the mutation of CheatSheetList, 
+and then DataFileWriter will use the information in CheatSheetList to save it to the memory in the form of a txt file.
+
+![](Images/Image10.PNG)
+
+Image 8: Class diagram of CheatSheetList class
+
+The cheatSheets ArrayList is of type private, and setter/getter methods must be invoked to access the list. 
+For these setter and getter methods, it accepts both accessing by index and accessing by name to allow the user to easily search and fetch the cheat sheet. 
+When /list or /find command is invoked, printCheatSheetNames will return a string consisting all relevant cheat sheets to be printed by Printer class.
+
+
+#### 3.2.5. Data Storage (Theo)
+
+This feature allows the application to read and update data in the form of text files. 
+Having an external source to store data will allow the application to be able to retrieve it when it is relaunched at another instance. 
+This prevents the user from having to repeatedly create new cheatsheets and update the application settings each time he opens up CheatLogs. 
+
+
+This feature is split into 3 functions, which are indicated in Table 1 below.
```

Comment:
There seems to be some issue with the markdown script for the table as it does not display a table on the DG. Seems to be the case that there's a whitespace missing between the '|' character and the text in each cell of the table.